src/main/java/org/testng/internal/Graph.java

```diff
@@ -121,7 +122,10 @@ public void topologicalSort() {
     // Sort the nodes alphabetically to make sure that methods of the same class
     // get run close to each other as much as possible
     //
-    Collections.sort(nodes2);
+    Comparator<Node> comparator = Systematiser.getComparator();
```

Comment:
Ok, I understood. But why would the `comparator` instantiation be plugged into a `Graph` instance from outside ? The `comparator` instantiation is completely dependent on just a JVM argument and as such I thought we can have the `comparator` be a field within `Graph` but skip the constructor parameter part.